### PR TITLE
Add Windows and MacOS CI

### PR DESCRIPTION
**Pull request by xuhdev**
We currently allow Windows CI to fail because there seems to be some real bugs, and we want to be able to run the CI every time we fix a bug.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    - TOXENV: py36
+      APPVEYOR_BUILD_WORKER_IMAGE: "Visual Studio 2013"
+    - TOXENV: py37
+      # Because MacOS and Linux are quite similar for our purposes, we only test one Python version on MacOS
+      # to avoid surprise and conserve resource (we have only 1 test running concurrently on AppVeyor).'
+
+      # We put this test right after the first Windows test because we only have 1 test running concurrently, and this
+      # is the test that is the most likely to fail after the first Windows test has passed, because it is very
+      # different from Windows.
+      APPVEYOR_BUILD_WORKER_IMAGE: macOS
```
Comment (edwardleardi) [inline appveyor.yml]: Makes sense, I take it running concurrently drains credits faster or is a premium feature of appveyor?

**Review by edwardleardi — COMMENTED** [review]
Can you add a quick explanation to CONTRIBUTING.md about how we use Travis for Linux and AppVeyor for Windows + Mac? Everything else looks good.

File: appveyor.yml (hunk 1 above)
Comment (xuhdev) [reply appveyor.yml]: It's a premium feature and we are on the free plan

**Review by edwardleardi — APPROVED** [review]
LGTM

**Pull request merged**